adapters/grid/grid.go

```diff
@@ -333,19 +345,23 @@ func (a *GridAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReq
 	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
 		return nil, []error{err}
 	}
+	var bidRespWithType GridResponseType
+	if err := json.Unmarshal(response.Body, &bidRespWithType); err != nil {
+		return nil, []error{err}
+	}
```

Comment:
You are unmarshaling `response.Body` twice here, which is rather inefficient. You may be able to get around it with embedding.

`type GridBid struct {
         *openrtb.Bid
	ContentType openrtb_ext.BidType `json:"content_type"`
}

type GridSeatBid struct {
        *openrtb.SeatBid
	Bid []GridBid `json:"bid"`
}

type GridResponse struct {
         *openrtb.BidResponse
	SeatBid []GridSeatBid `json:"seatbid,omitempty"`
}
`
This way `GridResponse` has all the openrtb fields, but `Bid` is overridden to include the `content_type` field. Thus you would only need to unmarshal the response once and use the same struct for everything,

Reply:
Oh, yes, I did not consider that you need to get back to the original `openrtb2.Bid`. As long as it is just the `Bid` you need to recast, I believe `openrtb2Bid := sb.Bid[i].Bid` will do it, assuming that `sb` is your `GridSeatBid` type.

Reply:
Thank you, your tip was very helpful!